### DISCUSSION TITLE
Feature idea: conditionally show job parameter UI elements based on other parameter values

### DISCUSSION THREAD

**Discussion by crowecawcaw** [OpenJobDescription/openjd-specifications#42] [Ideas]
Example: you have a job template that renders a scene across a frame range. The frame range is saved inside the scene itself, but you want to allow the person submitting the job to be able to optionally override the frame range settings when submitting the OpenJD job.

With the current OJD UI hints, you can add a checkbox for "Override frame range" and an input for "Frame range", but the "Frame range" input will always be visible even when "Override frame range" is not selected. Ideally, the "Frame range" input could be configured to only show when the override is enabled.

As design inspiration, Houdini's UI language for plugins [supports conditional UI elements](https://www.sidefx.com/docs/houdini/ref/windows/optype.html#conditions) and [here's an example](https://github.com/aws-deadline/deadline-cloud-for-houdini/blob/mainline/src/deadline/houdini_submitter/otls/deadline_cloud.hda/Driver_1deadline__cloud/DialogScript#L673) of that syntax being used in Deadline Cloud's Houdini submitter.

What do you all think about adding a similar feature to OJD's UI hints?

**Comment by mwiebe** [comment]
This idea could build on https://github.com/OpenJobDescription/openjd-specifications/discussions/79 that proposes to add expression support inside the template format string `{{...}}` syntax.
